### DISCUSSION TITLE
Welcome to PasswordPusher Discussions!

### DISCUSSION THREAD

**Discussion by pglombardo** [pglombardo/PasswordPusher#226] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dbuehler77** [comment]
@pglombardo, what's the recommended way to upgrade/update PWPush? We run our own internal server.

**Comment by pglombardo** [comment]
Are you running from source?  Then:

```bash
git pull # get latest code
bundle install # install all required packages
yarn install # install moar required packages :-)
rails webpacker:compile # compile assets
rails db:migrate # migrate database to latest schema
```
and restart the application.

If you are running a docker container, the `release` tag (e.g. [pglombard/pwpush-postgres:release](https://hub.docker.com/layers/pglombardo/pwpush-postgres/release/images/sha256-cb0de6aa48ede2a8846be8b5281f1dc481a0e4078a9682e8db1d3b7ea5ac59f9?context=repo)) always points to the latest stable release so an occasional re-fetch of that tag will update your instance automatically (and run all those steps automatically).

Does this cover how you are running your internal version?
